eth/eventdatahandler/event_data_handler_test.go

```diff
@@ -0,0 +1,171 @@
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	edh, err := setupDataHandler(t, ctx, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+	eventsCh := make(chan ethtypes.Log)
+	go func() {
+		defer close(eventsCh)
+		for _, event := range events {
+			eventsCh <- event
+		}
+	}()
+	lastProcessedBlock, err := edh.HandleBlockEventsStream(eb.BatchEvents(eventsCh), false)
+	require.Equal(t, uint64(0x89EBFF), lastProcessedBlock)
+	require.NoError(t, err)
+}
+
+func setupDataHandler(t *testing.T, ctx context.Context, logger *zap.Logger) (*EventDataHandler, error) {
+	options := basedb.Options{
+		Type:       "badger-memory",
+		Path:       "",
+		Reporting:  true,
+		GCInterval: 0,
+		Ctx:        ctx,
+	}
+
+	db, err := kv.New(logger, options)
+	if err != nil {
+		return nil, err
+	}
+
+	eventDB := eventdb.NewEventDB(db.Badger())
+	storageMap := ibftstorage.NewStores()
+	nodeStorage, operatorData := setupOperatorStorage(logger, db)
+	keyManager, err := ekm.NewETHKeyManagerSigner(logger, db, networkconfig.NetworkConfig{}, true)
+	if err != nil {
+		return nil, err
+	}
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	bc := beacon.NewMockBeaconNode(ctrl)
+	validatorCtrl := validator.NewController(logger, validator.ControllerOptions{
+		Context:         ctx,
+		DB:              db,
+		RegistryStorage: nodeStorage,
+	})
+
+	cl := executionclient.New("test", common.Address{})
+
+	edh, err := New(
+		eventDB,
+		cl,
+		validatorCtrl,
+		operatorData,
+		nodeStorage.GetPrivateKey,
+		keyManager,
+		bc,
+		storageMap,
+		WithFullNode(),
+		WithLogger(logger))
+	if err != nil {
+		return nil, err
+	}
+	return edh, nil
+}
+
+func setupOperatorStorage(logger *zap.Logger, db basedb.IDb) (operatorstorage.Storage, *registrystorage.OperatorData) {
+	nodeStorage, err := operatorstorage.NewNodeStorage(logger, db)
+	if err != nil {
+		logger.Fatal("failed to create node storage", zap.Error(err))
+	}
+	operatorPubKey, err := nodeStorage.SetupPrivateKey(logger, "", true)
+	if err != nil {
+		logger.Fatal("could not setup operator private key", zap.Error(err))
+	}
+
+	_, found, err := nodeStorage.GetPrivateKey()
+	if err != nil || !found {
+		logger.Fatal("failed to get operator private key", zap.Error(err))
+	}
+	var operatorData *registrystorage.OperatorData
+	operatorData, found, err = nodeStorage.GetOperatorDataByPubKey(logger, operatorPubKey)
+	if err != nil {
+		logger.Fatal("could not get operator data by public key", zap.Error(err))
+	}
+	if !found {
+		operatorData = &registrystorage.OperatorData{
+			PublicKey: operatorPubKey,
+		}
+	}
+
+	return nodeStorage, operatorData
+}
+
+func unmarshalLog(t *testing.T, rawOperatorAdded string) *types.Log {
+	var vLogOperatorAdded types.Log
+	err := json.Unmarshal([]byte(rawOperatorAdded), &vLogOperatorAdded)
+	require.NoError(t, err)
+	contractAbi, err := abi.JSON(strings.NewReader(contract.ContractABI))
+	require.NoError(t, err)
+	require.NotNil(t, contractAbi)
+	return &vLogOperatorAdded
```

Comment:
It gets dereferenced afterwards in all places where `unmarshalLog` is called. Can we return just `vLogOperatorAdded` instead?